packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -555,12 +555,29 @@ export class FeatureDevController {
                         result = MetricDataResult.Fault
                     }
                     break
+                case MonthlyConversationLimitError.name:
+                case CodeIterationLimitError.name:
                 case PromptRefusalException.name:
                 case NoChangeRequiredException.name:
                     result = MetricDataResult.Error
                     break
                 default:
-                    result = MetricDataResult.Fault
+                    const startCodeGenClientError = ['Improperly formed request', 'Resource not found']
```

Comment:
Can we store these errors someone else then buried in the PR? I see "Improperly formed request" is used here but also in the controller test as well